mautic/mautic
# grapejs and ckeditor update

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [x]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ] no breaking changes will work with v4 and v5 without issue.
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

Fixes a few issues such as image resizing on paste to respect the container it is being pasted in.
copy paste of text was moving the page down on each paste this also contains the fix for that,
upgraded the use of ckeditor to latest version, we also use typescript to load it into grapejs.

We also added a file manager so help with searching for assets

credit for this goes to @paulRose

#### Steps to test this PR:
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Launch the page builder and this PR will fix a copy paste issue that we have seen where pasting caused the page to move itself down to the bottom of the view.
3. Code mode should be working again.
4. ckeditor will be upgraded and should work as expected.
